### PR TITLE
[util/design] Encode bit width in OTP vmem filename

### PR DESCRIPTION
**Pull request by dmcardle**
See #10867

### CODE REVIEW

File: rules/autogen.bzl [hunk 1]
```diff
@@ -105,7 +105,7 @@ autogen_chip_info = rule(
 )
 
 def _otp_image(ctx):
-    output = ctx.actions.declare_file(ctx.attr.name + ".vmem")
+    output = ctx.actions.declare_file(ctx.attr.name + "-24.vmem")
```
[the remark this answers is not in the document]
Comment (dmcardle) [reply rules/autogen.bzl]: [Moved to a TODO in rules/autogen.bzl]

**Comment by dmcardle** [comment]
@msfschaffner ping :)

**Comment by msfschaffner** [comment]
Sorry about that, will take a look. 
Also, would you mind rebasing this onto TOT again in order to get the latest CI fixes?

**Comment by dmcardle** [comment]
Thanks! I rebased right before my last message, I think that's up to date with master right now.

**Comment by msfschaffner** [comment]
Hm looks like there not all clang format issues have been fixed yet. Nevermind (we'll just have to rebase this again once it is fixed).

File: util/design/lib/OtpMemImg.py [hunk 2]
```diff
@@ -76,7 +76,8 @@ def _present_64bit_digest(data_blocks, iv, const):
     return state
 
 
-def _to_hexfile_with_ecc(data, annotation, config, data_perm):
+def _to_hexfile_with_ecc(data, annotation, config,
+                         data_perm) -> tuple[str, int]:
```
[the remark this answers is not in the document]
Comment (dmcardle) [reply util/design/lib/OtpMemImg.py]: Done in 858e13b25bd1f7bc57a1452d30162eb25dc37f09. (And rebased for the clang-format fix!)

**Review by msfschaffner — APPROVED** [review]
LGTM

**Comment by msfschaffner** [comment]
Jup, will do once FPGA passes.

**Pull request merged**